gateway/handler_codec.go

```diff
@@ -69,10 +69,20 @@ func (i *handler) serveCodec(ctx context.Context, w http.ResponseWriter, r *http
 	defer data.Close()
 
 	setIpfsRootsHeader(w, rq, &pathMetadata)
-	return i.renderCodec(ctx, w, r, rq, data)
+
+	blockSize, err := data.Size()
+	if !i.handleRequestErrors(w, r, rq.contentPath, err) {
+		return false
+	}
+
+	if !i.seekToStartOfFirstRange(w, r, data) {
+		return false
+	}
```

Comment:
@hacdias I don't know if you or I put this in, but this looks  suspicious. Why would we be doing manipulation of the data range before doing the full rendering/transformation?

Reply:
The blame says you added it - tbf, I do not remember adding it, so it's probably right. (698eccfdefb70d57fc5cab8cbf36f47afb00b25b). I have no idea about what it might be doing.